src/APIFunctions/2DPrinting.js

```diff
@@ -0,0 +1,56 @@
+import axios from 'axios'
+
+export const range = (start, end) => {
+  const length = end - start
+  return Array.from({ length }, (_, i) => start + i)
+}
+
+export function parseRange (pages, maxPages) {
+  var result = new Set()
+  var subr = pages.split(',')
+  subr.forEach(element => {
+    const x = element.split('-')
+    const arr = range(Number(x[0]), Number(x[x.length - 1]) + 1)
+    arr.forEach(element => {
+      result.add(element)
+    })
+  })
+  result.delete(0)
+  result.forEach(element => {
+    if (element > maxPages) result.delete(element)
+  })
+  if (result.size === 0) {
+    var arr = new Set(range(1, maxPages + 1))
+    return arr
+  }
+  return result
+}
+
+export function printPage (data) {
+  axios
+    .post('/api/print/submit', data)
+    .then(res => {
+      console.log(res)
+    })
+    .catch(() => {
+      window.alert('You do not have enough pages!')
```

Comment:
instead, show a modal or red text to the user, [this answer](https://ux.stackexchange.com/a/4522) provides some insight on why we shouldnt use alerts.

Reply:
maybe send 'true/false' to enable/disable the modal